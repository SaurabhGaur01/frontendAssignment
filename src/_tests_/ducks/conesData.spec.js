import {
    setConesData,
    conesDataReducer,
} from '../../ducks/conesData';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('conesData Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setConesData should match the snapshot' , () => {
            expect(setConesData(mockObject)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(conesDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(conesDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(conesDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value of set data' , () => {
            expect(conesDataReducer('', setConesData(mockObject))).toEqual(mockObject);
        });
    });
});