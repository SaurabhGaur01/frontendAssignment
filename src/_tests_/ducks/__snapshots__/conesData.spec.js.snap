// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conesData Duck Tests Action Generator setConesData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/app/SET_CONES_DATA",
}
`;
